parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -280,6 +283,90 @@ WriteBuilder withWriterVersion(WriterVersion version) {
             metricsConfig);
       }
     }
+
+    private static class Context {
+      private final int rowGroupSize;
+      private final int pageSize;
+      private final int dictionaryPageSize;
+      private final CompressionCodecName codec;
+      private final String compressionLevel;
+
+      private Context(int rowGroupSize, int pageSize, int dictionaryPageSize,
+                      CompressionCodecName codec, String compressionLevel) {
+        this.rowGroupSize = rowGroupSize;
+        this.pageSize = pageSize;
+        this.dictionaryPageSize = dictionaryPageSize;
+        this.codec = codec;
+        this.compressionLevel = compressionLevel;
+      }
+
+      public static Context dataContext(Map<String, String> config) {
+        int rowGroupSize = Integer.parseInt(config.getOrDefault(
+            PARQUET_ROW_GROUP_SIZE_BYTES, PARQUET_ROW_GROUP_SIZE_BYTES_DEFAULT));
+
+        int pageSize = Integer.parseInt(config.getOrDefault(
+            PARQUET_PAGE_SIZE_BYTES, PARQUET_PAGE_SIZE_BYTES_DEFAULT));
+
+        int dictionaryPageSize = Integer.parseInt(config.getOrDefault(
+            PARQUET_DICT_SIZE_BYTES, PARQUET_DICT_SIZE_BYTES_DEFAULT));
```

Comment:
Nit:  could we align this value parser with the following `deleteContext` by using `PropertyUtil.propertyAsInt` ?

Reply:
This is what I wanted to use too. Unfortunately, `PARQUET_ROW_GROUP_SIZE_BYTES_DEFAULT` is actually a String. That's why it is not possible to use `PropertyUtil`. Copied this code as it was in the original implementation.